OBSDOCS-960: Edit of TRACING-3981 Add OpenTelemetry Kubernetes events receiver (#75019)
openshift/openshift-docs: modules/otel-collector-components.adoc

[id="otlp-receiver_{context}"]
=== OTLP Receiver

> The OTLP Receiver ingests traces and metrics using the OpenTelemetry protocol (OTLP).

Reviewer: ```suggestion
The OTLP Receiver ingests traces and metrics by using the OpenTelemetry protocol (OTLP).
```
Good to have (If you did care to fix this)

Author: Thank you. TBH, I normally include `by` in such phrases. However, here the OTLP is not a mean but rather an attribute. So it seemed to appropriate for the intended meaning to have `by` omitted.

----
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRole
metadata:
  name: otel-collector
  labels:
    app: otel-collector
rules:
- apiGroups:
  - ""
  resources:
  - events
  - namespaces
  - namespaces/status
  - nodes
  - nodes/spec
  - pods
  - pods/status
  - replicationcontrollers
  - replicationcontrollers/status
  - resourcequotas
  - services
  verbs:
  - get
  - list
  - watch
- apiGroups:
  - apps
  resources:
  - daemonsets
  - deployments
  - replicasets
  - statefulsets
  verbs:
  - get
  - list
  - watch
- apiGroups:
  - extensions
  resources:
  - daemonsets
  - deployments
  - replicasets
  verbs:
  - get
  - list
  - watch
- apiGroups:
  - batch
  resources:
  - jobs
  - cronjobs
  verbs:
  - get
  - list
  - watch
- apiGroups:
    - autoscaling
  resources:
    - horizontalpodautoscalers
  verbs:
    - get
    - list
    - watch
----

.OpenTelemetry Collector custom resource with the enabled Kubernetes Event Receiver
[source,yaml]
----
  serviceAccount: otel-collector # <1>
  config: |
    receivers:
      k8s_events:
        namespaces: [project1, project2] # <2>
    service:
      pipelines:
        logs:
          receivers: [k8s_events]
----

> <1> The service account of the Collector that has the required ClusterRole `otel-collector` RBAC.

Reviewer: ```suggestion
<1> The service account of the Collector that has the required cluster role `otel-collector` RBAC.
```

Reviewer: Good to have ..

Author: I looked it up at the time. "ClusterRole" is used in the OpenShift docs as-is. For example: https://docs.openshift.com/container-platform/4.14/rest_api/rbac_apis/clusterrole-rbac-authorization-k8s-io-v1.html

> <2> The list of namespaces to collect events from. Defaults to empty, which means all namespaces.

Reviewer: ```suggestion
<2> The list of namespaces to collect events from. The default value is empty, which means that all namespaces are collected.
```

Author: Thank you. Applying your suggestion.

|===

[id="resource-detection-processor_{context}"]
=== Resource Detection Processor

The Resource Detection Processor is currently a link:https://access.redhat.com/support/offerings/techpreview[Technology Preview] feature only.

> The Resource Detection Processor identifies host resource details in alignment with OpenTelemetry's resource semantic standards. Using the detected information, it can add or replace the resource values in telemetry data. This processor supports traces, metrics, and can be used with multiple detectors such as the Docket metadata detector or the `OTEL_RESOURCE_ATTRIBUTES` environment variable detector.

Reviewer: ```suggestion
The Resource Detection Processor identifies host resource details in alignment with OpenTelemetry's resource semantic standards. By using the detected information, it can add or replace the resource values in telemetry data. This processor supports traces, metrics, and you can use it with multiple detectors such as the Docket metadata detector or the `OTEL_RESOURCE_ATTRIBUTES` environment variable detector.
```

Reviewer: Good to have (if you did care to fix it)

Author: Thank you for the suggestion. I didn't add "By" to "Using the detected information" because the "detected information" is not a tool that performs the action (compared to say "By using the processor").

Hash:: Hashes an existing attribute value as SHA1.

> Extract:: Extracts values by using a regular expression rule from the input key to the target keys defined in the rule. If a target key already exists, it will be overridden similarly to the Span Processor's `to_attributes` setting with the existing attribute as the source.

Reviewer: ```suggestion
Extract:: Extracts values by using a regular expression rule from the input key to the target keys defined in the rule. If a target key already exists, it is overridden similarly to the Span Processor's `to_attributes` setting with the existing attribute as the source.
```

Reviewer: Good to have ..

Author: Thank you for spotting this, applying.

Attributes represent the actions that are applied to the resource attributes, such as delete the attribute, insert the attribute, or upsert the attribute.

[id="span-processor_{context}"]
=== Span Processor

The Span Processor is currently a link:https://access.redhat.com/support/offerings/techpreview[Technology Preview] feature only.

> The Span Processor modifies the span name based on its attributes or extracts the span attributes from the span name. It can also change the span status. It can also include or exclude spans. This processor supports traces.

Reviewer: ```suggestion
The Span Processor modifies the span name based on its attributes or extracts the span attributes from the span name. It can also change the span status, and include or exclude spans. This processor supports traces.
```

Reviewer: Good to have ..

Author: Thank you, applying.

----

[id="filter-processor_{context}"]
=== Filter Processor

The Filter Processor is currently a link:https://access.redhat.com/support/offerings/techpreview[Technology Preview] feature only.

> The Filter Processor leverages the OpenTelemetry Transformation Language to establish criteria for discarding telemetry data. If any of these conditions are satisfied, the telemetry data are discarded. The conditions can be combined by using the logical OR operator. This processor supports traces, metrics, and logs.

Reviewer: ```suggestion
The Filter Processor leverages the OpenTelemetry Transformation Language to establish criteria for discarding telemetry data. If any of these conditions are satisfied, the telemetry data is discarded. You can combine the conditions by using the logical OR operator. This processor supports traces, metrics, and logs.
```

Reviewer: Good to have ..

Author: I'd like to argue that "telemetry data" is plural. Rewriting the passive voice.

<3> Filters the spans that have the `host.name == localhost` resource attribute.

[id="routing-processor_{context}"]
=== Routing Processor

The Routing Processor is currently a link:https://access.redhat.com/support/offerings/techpreview[Technology Preview] feature only.

> The Routing Processor routes logs, metrics, or traces to specific exporters. This processor can read a header from an incoming HTTP request (gRPC or plain HTTP) or can read a resource attribute, and then directs the trace information to relevant exporters according to the read value.

Reviewer: ```suggestion
The Routing Processor routes logs, metrics, or traces to specific exporters. This processor can read a header from an incoming gRPC or plain HTTP request, or can read a resource attribute, and then directs the trace information to relevant exporters according to the read value.
```

Reviewer: Good to have ..

Author: Thank you. I edited this even further.

Exporters send data to one or more back ends or destinations.

[id="otlp-exporter_{context}"]
=== OTLP Exporter

> The OTLP gRPC Exporter exports traces and metrics using the OpenTelemetry protocol (OTLP).

Reviewer: ```suggestion
The OTLP gRPC Exporter exports traces and metrics by using the OpenTelemetry protocol (OTLP).
```

Reviewer: Good to have ..

Author: Thank you. Applied.

<7> Headers are sent for every request performed during an established connection.

[id="otlp-http-exporter_{context}"]
=== OTLP HTTP Exporter

> The OTLP HTTP Exporter exports traces and metrics using the OpenTelemetry protocol (OTLP).

Reviewer: ```suggestion
The OTLP HTTP Exporter exports traces and metrics by using the OpenTelemetry protocol (OTLP).
```

Reviewer: Good to have ..

Author: Thank you. Applying.

<9> Adds the metrics types and units suffixes. Must be disabled if the monitor tab in Jaeger console is enabled. The default is `true`.

[id="kafka-exporter_{context}"]
=== Kafka Exporter

The Kafka Exporter is currently a link:https://access.redhat.com/support/offerings/techpreview[Technology Preview] feature only.

> The Kafka Exporter exports logs, metrics, and traces to Kafka. This exporter uses a synchronous producer that blocks and does not batch messages. It must be used with batch and queued retry processors for higher throughput and resiliency.

Reviewer: ```suggestion
The Kafka Exporter exports logs, metrics, and traces to Kafka. This exporter uses a synchronous producer that blocks and does not batch messages. You must use it with batch and queued retry processors for higher throughput and resiliency.
```

Reviewer: Good to have ..

Author: Thank you. Applying.

[id="jaegerremotesampling-extension_{context}"]
=== Jaeger Remote Sampling Extension

The Jaeger Remote Sampling Extension is currently a link:https://access.redhat.com/support/offerings/techpreview[Technology Preview] feature only.

> The Jaeger Remote Sampling Extension allows serving sampling strategies after Jaeger's remote sampling API. You can configure this extension to proxy requests to a backing remote sampling server such as a Jaeger collector down the pipeline or to a static JSON file from the local file system.

Reviewer: ```suggestion
The Jaeger Remote Sampling Extension enables serving sampling strategies after Jaeger's remote sampling API. You can configure this extension to proxy requests to a backing remote sampling server such as a Jaeger collector down the pipeline or to a static JSON file from the local file system.
```

Reviewer: Good to have ..

Author: Thank you. Applying.

[id="pprof-extension_{context}"]
=== Performance Profiler Extension

The Performance Profiler Extension is currently a link:https://access.redhat.com/support/offerings/techpreview[Technology Preview] feature only.

> The Performance Profiler Extension enables the Go `net/http/pprof` endpoint. This is typically used by developers to collect performance profiles and investigate issues with the service.

Reviewer: ```suggestion
The Performance Profiler Extension enables the Go `net/http/pprof` endpoint. Developers use this to collect performance profiles and investigate issues with the service.
```

Reviewer: Good to have ..

Author: Thank you. I edited it even further.
